tests/test_omegaconf.py

```diff
@@ -518,3 +518,66 @@ def test_resolve(cfg: Any, expected: Any) -> None:
 def test_resolve_invalid_input() -> None:
     with raises(ValueError):
         OmegaConf.resolve("aaa")  # type: ignore
+
+
+@mark.parametrize(
+    ("cfg_params", "clear_resolver_params", "expected"),
+    [
+        # tuple
+        param(
+            # cfg_params
+            dict(
+                name="iamnew",
+                resolver=lambda x: str(x).lower(),
+                use_cache=False,
+                replace=False,
+            ),
+            # clear_resolver_params
+            dict(name="iamnew"),
+            # expected
+            dict(pre_addition=False, post_addition=True),
+            id="remove-new-custom-resolver",
+        ),
+        # tuple
+        param(
+            # cfg_params: Check with a default resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="oc.env"),
+            # expected
+            dict(pre_addition=True, post_addition=True),
+            id="remove-default-resolver",
+        ),
+        # tuple
+        param(
+            # cfg_params: Check with a nonexisting resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="idonotexist"),
+            # expected
+            dict(pre_addition=False, post_addition=False),
+            id="remove-nonexistent-resolver",
+        ),
+    ],
+)
+def test_clear_resolver(
+    restore_resolvers: Any, cfg_params: Any, clear_resolver_params: Any, expected: Any
```

Comment:
Since `clear_resolver_params` is a dict whose only key is `"name"`, it would be simpler to just pass in the string as an argument to the test function:

```python
def test_clear_resolver(restore_resolvers: Any, cfg_params: Any, name: str, expected: Any) -> None:
```

Reply:
Done!